pinot-controller/src/main/java/com/linkedin/pinot/controller/api/storage/PinotFS.java

```diff
@@ -0,0 +1,98 @@
+/**
```

Comment:
Partial review - have only looked at this interface.

- How did we decide on the set of operations to be supported for the FS interface? For ex, I don't see mkdir/s or method to get the status of a file etc.

Reply:
I came up with it based on what I thought pinot needed; I can relook at all the fs methods to see if any others could be useful.